### ISSUE TITLE
cmake - I3_TESTDATA defaults are set after enable_testing() (Trac #1568)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1560]
`I3_TESTDATA` is used to switch on testing (with `ctest`) in `cmake/toplevel.cmake`, but its defaults are set later in `cmake/config.cmake`. If using SYSTEM_PACKAGES and not CVMFS, and `I3_TESTDATA` isn't set, testing won't turn on when you are expecting it to.

Warn the user, or move the location of where the defaults are set.

Migrated from https://code.icecube.wisc.edu/ticket/1568
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:15:12", 
    "description": "`I3_TESTDATA` is used to switch on testing (with `ctest`) in `cmake/toplevel.cmake`, but its defaults are set later in `cmake/config.cmake`. If using SYSTEM_PACKAGES and not CVMFS, and `I3_TESTDATA` isn't set, testing won't turn on when you are expecting it to.\n\nWarn the user, or move the location of where the defaults are set.", 
    "reporter": "nega", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547252112463010", 
    "component": "cmake", 
    "summary": "cmake - I3_TESTDATA defaults are set after enable_testing()", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-02-25T21:54:16", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160422T14:04:00: 
* nega commented:

> In changeset 2523:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2523"
> group I3_TESTDATA stuff together in toplevel.cmake
> 
> closes #1560
> ```


* nega changed status from "new" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:15:12: nwhitehorn commented:

> In changeset 170246:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170246"
> group I3_TESTDATA stuff together in toplevel.cmake
> 
> closes #1560
> ```
